Adding the `--no-production` option to `preact build` produces a Pre-Render/Server-Side-Render error.

<!-- Please don't delete this template -->
<!-- Before creating an issue please make sure you are using the latest version of preact-cli. -->

**Do you want to request a *feature* or report a *bug*?**

_Report a bug_

**What is the current behaviour?**

`preact build --no-production` causes a server-side exception. `ReferenceError: window is not defined`.

### Full Error Message
```
Unable to read sourcemap: /Users/johnhaitas/src/johnhaitas/preact-cli-no-production-ssr/build/ssr-build/ssr-bundle.js.map

ReferenceError: window is not defined
method: null
[object Object]
This is most likely caused by using DOM or Web APIs.
Pre-render runs in node and has no access to globals available in browsers.

Consider wrapping code producing error in: 'if (typeof window !== "undefined") { ... }'

Alternatively use 'preact build --no-prerender' to disable prerendering.

See https://github.com/developit/preact-cli#pre-rendering for further information.npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! preact-cli-no-production-ssr@0.0.0 build-dev: `preact build --no-production`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the preact-cli-no-production-ssr@0.0.0 build-dev script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     /Users/johnhaitas/.npm/_logs/2018-10-03T18_50_35_625Z-debug.log
```

[2018-10-03T18_50_35_625Z-debug.log](https://github.com/developit/preact-cli/files/2443436/2018-10-03T18_50_35_625Z-debug.log)

The offending line comes from `style-loader` https://github.com/webpack-contrib/style-loader/blob/v0.18.2/lib/addStyles.js#L23.

**If the current behaviour is a bug, please provide the steps to reproduce.**

I have created a repo which adds a `build-dev` npm run script
https://github.com/johnhaitas/preact-cli-no-production-ssr

Reproduce the bug by running `npm run build-dev`

**What is the expected behaviour?**

The server-side pre-render should be successful.

**Please mention other relevant information.**
- Node version - `v8.12.0`
- npm version - `6.4.1`
- Operating system - `MacOS High Sierra`
- CLI version - `2.2.1`

--- a/src/commands/build.js
+++ b/src/commands/build.js
@@ -4,6 +4,8 @@ import { isDir, error } from '../util';
 import asyncCommand from '../lib/async-command';
 import runWebpack, { showStats, writeJsonStats } from '../lib/webpack/run-webpack';
 
+const toBool = val => val === void 0 || (val === 'false' ? false : val);
+
 export default asyncCommand({
 	command: 'build [src] [dest]',
 
@@ -59,6 +61,9 @@ export default asyncCommand({
 	async handler(argv) {
 		let cwd = resolve(argv.cwd);
 		let modules = resolve(cwd, 'node_modules');
+
+		argv.prerender = toBool(argv.prerender);
+		argv.production = toBool(argv.production);
 
 		if (!isDir(modules)) {
 			return error('No `node_modules` found! Please run `npm install` before continuing.', 1);

--- a/src/lib/webpack/run-webpack.js
+++ b/src/lib/webpack/run-webpack.js
@@ -13,6 +13,7 @@ import { error, isDir, warn } from '../../util';
 
 export default function (watch=false, env, onprogress) {
 	env.isProd = env.production; // shorthand
+	env.isWatch = !!watch; // use HMR?
 	env.cwd = resolve(env.cwd || process.cwd());
 
 	// env.src='src' via `build` default

--- a/src/lib/webpack/webpack-base-config.js
+++ b/src/lib/webpack/webpack-base-config.js
@@ -25,7 +25,7 @@ function resolveDep(dep, cwd) {
 }
 
 export default function (env) {
-	const { cwd, isProd, src, source } = env;
+	const { cwd, isProd, isWatch, src, source } = env;
 
 	// Apply base-level `env` values
 	env.dest = resolve(cwd, env.dest || 'build');
@@ -210,7 +210,7 @@ export default function (env) {
 			// Extract CSS
 			new ExtractTextPlugin({
 				filename: isProd ? 'style.[contenthash:5].css' : 'style.css',
-				disable: !isProd,
+				disable: isWatch,
 				allChunks: true
 			}),
 			new webpack.optimize.CommonsChunkPlugin({
@@ -239,7 +239,7 @@ export default function (env) {
 			})
 		] : []),
 
-		devtool: isProd ? 'source-map' : 'cheap-module-eval-source-map',
+		devtool: isWatch ? 'cheap-module-eval-source-map' : 'source-map',
 
 		node: {
 			console: false,

